sys/include/xtimer/implementation.h

```diff
@@ -111,22 +111,23 @@ void _xtimer_tsleep(uint32_t offset, uint32_t long_offset);
 
 static inline uint32_t _xtimer_now(void)
 {
-#if XTIMER_MASK
-    uint32_t latched_high_cnt, now;
-
-    /* _high_cnt can change at any time, so check the value before
-     * and after reading the low-level timer. If it hasn't changed,
-     * then it can be safely applied to the timer count. */
+    uint32_t now, elapsed;
 
-    do {
-        latched_high_cnt = _xtimer_high_cnt;
-        now = _xtimer_lltimer_now();
-    } while (_xtimer_high_cnt != latched_high_cnt);
-
-    return latched_high_cnt | now;
+    /* time sensitive since _long_cnt and _xtimer_current_time are updated here */
+    uint8_t state = irq_disable();
+    now = _xtimer_lltimer_now();
+#if XTIMER_MASK
+    elapsed = _xtimer_lltimer_mask(now - _xtimer_lltimer_mask((uint32_t)_xtimer_current_time));
```

Comment:
There was a bug here: it was before "elapsed = now - _xtimer_lltimer_mask((uint32_t)_xtimer_current_time);"